Polygon_mesh_processing/test/Polygon_mesh_processing/extrude_test.cpp

```diff
@@ -0,0 +1,74 @@
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/Polyhedron_3.h>
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Polygon_mesh_processing/extrude.h>
+#include <iostream>
+#include <fstream>
+
+typedef CGAL::Exact_predicates_inexact_constructions_kernel Kernel;
+typedef CGAL::Surface_mesh<Kernel::Point_3> SMesh;
+typedef CGAL::Polyhedron_3<Kernel> Polyhedron;
+
+template<typename MAP>
+struct Bot
+{
+  Bot(MAP map):map(map){}
+  template<typename VD, typename T>
+  void operator()(VD vd, const T&)
+  {
+    put(map, vd, get(map, vd)+Kernel::Vector_3(-2.0,0.0,1.0));
+  }
+  MAP map;
+  
+};
+
+template<typename MAP>
+struct Top
+{
+  Top(MAP map):map(map){}
+  
+  template<typename VD, typename T>
+  void operator()(VD vd, const T&)
+  {
+    put(map, vd, get(map, vd)+Kernel::Vector_3(0.0,2.0,-1.0));
```

Comment:
according to the doc, you are modifying the input here (maybe it is working in your example because vertices are ordered the same way).